### PR TITLE
factor in min option for maxIdle value

### PR DESCRIPTION
**Pull request by callmehiphop**
Fixes https://github.com/GoogleCloudPlatform/google-cloud-node/issues/2685

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/session-pool.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ function SessionPool(database, options) {
   options = options || {};
 
   this.database = database;
-  this.maxIdle = options.maxIdle || 1;
+  this.maxIdle = options.maxIdle || options.min || 1;
```
Comment (stephenplusplus) [inline src/session-pool.js]: Is this a place where `Math.min` would help?

Comment (stephenplusplus) [reply src/session-pool.js]: Or `max`?

**Comment by stephenplusplus** [comment]
@lukesneeringer should we wait to merge this for any reason?

File: src/session-pool.js (hunk 1 above)
Comment (lukesneeringer) [reply src/session-pool.js]:
```
> Math.min(undefined, 1)
<- NaN
```

So, no. :-)

Comment (stephenplusplus) [reply src/session-pool.js]:
```js
this.maxIdle = options.maxIdle /*0*/ || options.min /*10*/ || 1
// this.maxIdle = 10
```

Is this scenario realistic and assignment correct?

Comment (callmehiphop) [reply src/session-pool.js]: @stephenplusplus I just pushed a commit that checks to see if `options.maxIdle` is a number first.

**Comment by lukesneeringer** [comment]
Yeah, I want to merge repo-migration into it first so the tests will pass.

**Pull request merged**